### PR TITLE
Checkbox to disable adjudication.

### PR DESCRIPTION
**Pull request by vdbergh**
Lightly tested.

### CODE REVIEW

**Comment by vdbergh** [comment]
Lightly tested.

**Comment by vondele** [comment]
Thanks... I think it would be interesting to see how much longer games take if we get rid of adjudication altogether. I might do a local measurement.

**Comment by vdbergh** [comment]
Long ago somebody tested it. IIRC it wasn't dramatically more resource intensive.

**Comment by vondele** [comment]
so, local testing at 10+0.1s with and without adjudication:
```
average game duration adj.pgn :  27.6601 s Elo:  Elo difference: 26.1 +/- 2.6, LOS: 100.0 %, DrawRatio: 80.3 %
average game duration noadj.pgn :  31.4449 s Elo:  Elo difference: 25.6 +/- 2.8, LOS: 100.0 %, DrawRatio: 80.8 %
```
about 15% longer games without adjucation, but both Elo and drawratio seem very similar. 

So probably no reason to change the default, but having a checkbox is not too bad.

File: server/fishtest/views.py [hunk 1]
```diff
@@ -594,7 +597,7 @@ def strip_message(m):
         data["base_signature"] = data["new_signature"]
 
     for k, v in data.items():
-        if len(v) == 0:
+        if not k in ['adjudication'] and len(v) == 0:
```
Comment (ppigazzini) [inline server/fishtest/views.py:600]: `if k not in ['adjudication'] and len(v) == 0:`

**Comment by ppigazzini** [comment]
DEV updated, started a test disabling the adjudication.

File: server/fishtest/views.py (hunk 1 above)
Comment (vdbergh) [reply server/fishtest/views.py:600]: I guess they are equivalent but `not in` looks indeed nicer.

Comment (ppigazzini) [reply server/fishtest/views.py:600]: `not in` is an operator, always wondered why not a single word as `notin` or `nin`

**Comment by vdbergh** [comment]
It seems it's not working. The pgns contain adjudicate games despite Adjudication showing False on the test page https://dfts-0.pigazzini.it/tests/view/618aac4cfc741c9c217d13c9. Are you also running the updated worker? I did test this locally.

**Comment by vdbergh** [comment]
Tried it with my version of the worker. The pgn https://dfts-0.pigazzini.it/api/pgn/618aac4cfc741c9c217d13c9-382.pgn does not contain adjudicated games.

**Comment by ppigazzini** [comment]
Sorry, I missed the worker changes :(
Label added now.

**Comment by ppigazzini** [comment]
DEV and workers updated, started a couple of tests disabling adjudication.

**Comment by vdbergh** [comment]
Looked at some .pdfs. Now it appears to be working!

**Comment by ppigazzini** [comment]
Great! Unfortunately I'm a bit under pression with my job, very little time for testing/reviewing :(

**Comment by vdbergh** [comment]
> Great! Unfortunately I'm a bit under pression with my job, very little time for testing/reviewing :(

I recognize the situation :) However I think this PR isn't very risky. It is just passing a flag from the "new test page" to the worker along the standard path. We only have to make sure the flag arrives at its destination and is not inverted along the way.

**Review by ppigazzini — APPROVED** [review]

**Comment by ppigazzini** [comment]
All PRs merged, triggered the workers update, thank you @vdbergh :)

**Comment by vdbergh** [comment]
Thanks!

**Comment by vondele** [comment]
thank you both!

**Comment by vondele** [comment]
btw those changes would be roughly similar to what would be needed to support things like ponder or FRC.

**Pull request merged**